components/IngredientCards/IngredientCard.tsx

```diff
@@ -15,6 +15,11 @@ export const IngredientCard = (
     onDelete?(): void;
   }
 ) => {
+    // If the weight is not per g, we will display an equivalence
+    var weight = "";
+    if (props.quantity_default_weight_per_unit !== undefined) {
+      weight = " (" + (props.value! * props.quantity_default_weight_per_unit) + " g)";
```

Comment:
Not sure it's necessary, but you could be interested by the the JS [`toFixed()`](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed)

In order to avoid stuff like `3*0.1 = 0.30000000000000004`.

Reply:
It doesn't seem we need it, the only floating number we have is the "pinch" of salt which is 0.5, but computations seem to work well: 
![image](https://github.com/openfoodfacts/rate-my-recipe/assets/8158668/265aac36-8fad-4298-905a-f92637fde1b3)

Thanks for the link to toFixed(). The thing is we want to display 4.5g, but not 9.0g.

Reply:
I don't know JS method for that. Usually, I do 
- For one digit it's easy: `.toFixed(1).replace('.0', '')`
- For more than one it's harder `.toFixed(2).replace(\\.?0*$\, '')` see [regex101](https://regex101.com/r/T1loyc/1)